osscameroon/jobsika
# Add opencollective payment `POST /pay` endpoint

This pull request implements the `POST /pay` endpoint as part of https://github.com/osscameroon/jobsika/issues/366

Why ?
We want to allow jobsika users to create new job offers only once they have paid for the service.

Steps to verify:

- move to the `backend` folder with `cd ./backend`
- you need to get an api-key from open-collective, [find out more](https://blog.opencollective.com/open-collective-graphql-api-preview)
- cp `.opencollective-env-example .opencollective-env`
- add you api key to the `.opencollective-env`
- run `make start-api` to start the api
- then run the command `curl -vsS -X POST -H 'Content-Type: application/json' -d '{"email":"test@email.com", "tier": "je suis con", "job_offer_id": "1"}' http://localhost:7000/pay | jq` you should see something like this

```
{
  "tier_url": "https://opencollective.com/osscameroon/contribute/jobsika-joboffer-56908"
}
```

The `tier_url` is the link to the newly created payment tier.
